server/socket-manager.js

```diff
@@ -5,9 +5,18 @@ const getUserData = require('./database/get-user-data.js');
 const authenticateCookie = (socket, callback) => {
   const cookie = cookieParser.parse(socket.request.headers.cookie)['grow-user'];
   iron.unseal(cookie, process.env.COOKIE_PASSWORD, iron.defaults, (err, decodedCookie) => {
-    id = decodedCookie.id;
-    getUserData(id, (data) => {
-      socket.emit('userdata', data);
+    if(err) callback(err);
+    callback(null, decodedCookie.id);
+  });
+};
+
+const socketManager = (socket) => {
+  socket.on('authenticate', () => {
+    authenticateCookie(socket, (err, id) => {
+      if(err) socket.emit('User not authenticated');
+      getUserData(id, (data) => {
+        socket.emit('userdata', data);
+      });
```

Comment:
what are we doing with all this on the frontend? Are we still using the on connect event? Or is this replacing it?